### PR TITLE
Add ACM Installation Information

### LINKED ISSUE #229: ACM Install
@fritz-astronomer commented on [Wed May 12 2021](https://github.com/astronomer/docs-archived/issues/348)

**What is the purpose of the document/enhancement? Select all that apply:**

- [x] Information for an existing feature that isn't yet documented.
- [x] Additional considerations/ reference material for an existing feature or document.

**Problem/feature that requires documentation:**
Our ACM Certificate installation of Astronomer Enterprise isn't sufficiently documented anywhere. 

**Describe the solution you'd like**
A guide, probably referenced from the AWS install docs with: 
1) Provisioning and attaching a self-signed cert  (It isn't made clear anywhere that a self-signed cert **is still required**)
https://www.notion.so/astronomerio/Create-a-Self-Signed-Cert-and-Save-to-K8s-c435d83be6144616bd997d0c792eaf83

2) Fully outlining the nginx annotations required
https://astronomer.slack.com/archives/C01MCEU9EV7/p1620232918003300

**Are you the right person to review a draft of the document for technical accuracy? If not, who is?**
Potentially, I am working through it myself on a demo install and with customers, but am no expert.

---

@virajmparekh commented on [Tue Jul 06 2021](https://github.com/astronomer/docs-archived/issues/348#issuecomment-874985846)

@vishwas-astro this needs to be addressed; most AWS users would prefer this method for SSL as it's a more AWS native solution. There's nothing about this feature in our release notes or anywhere on our docs site, causing confusion for new members of the field teams and slower implementations. 

---

@vishwas-astro commented on [Tue Jul 06 2021](https://github.com/astronomer/docs-archived/issues/348#issuecomment-874997027)

@jwitz I dont know what else is in your plate, but is this something that you can work on over the next few days with some help from @fritz-astronomer and @pgagnon. 

---

@jwitz commented on [Tue Jul 06 2021](https://github.com/astronomer/docs-archived/issues/348#issuecomment-875006893)

@vishwas-astro I'm fully booked this week with the Beta rollout, would this involve just adding another option to [this section](https://www.astronomer.io/docs/enterprise/v0.25/install/aws/install-aws-standard#step-4-configure-tls) of our installation? 

I see we have written steps for the self-signed cert, but nothing on the actual ACM configuration required for this to happen. @fritz-astronomer If that's documented in Notion I can start working it into the guide, but I don't see this getting published until next week. I'm technically on vacation then, but online periodically to tie up loose ends on the Cloud Beta - could also do this if it's an absolute necessity.

---

@pgagnon commented on [Tue Jul 06 2021](https://github.com/astronomer/docs-archived/issues/348#issuecomment-875047646)

I'd hold off on publishing this for a little bit, because we're currently investigating issues with NLB deployed in private subnets.

cc @rbankston 

---

@vishwas-astro commented on [Wed Jul 07 2021](https://github.com/astronomer/docs-archived/issues/348#issuecomment-875825072)

@jwitz - you can work with @pgagnon and @rbankston to document this feature once they have completed some of their investigations here.

---

@jwitz commented on [Tue Aug 24 2021](https://github.com/astronomer/docs-archived/issues/348#issuecomment-904801667)

@pgagnon @rbankston  following up on this issue - did you investigate the issue with NLB, and is there anything else preventing us from getting this installation published?

### PR DESCRIPTION
**Pull request by kwcanuck**
Resolves #229 

@pgagnon when time permits, could you complete a review of this new information before a peer review is completed? Thanks!

### CODE REVIEW

File: software/install-aws-standard.md [hunk 1]
```diff
@@ -44,13 +45,13 @@ EKS is built off of Amazon's pre-existing EC2 service, so you can manage your Ku
 As you follow the guide linked above, keep in mind:
 
 * Each version of Astronomer Software is compatible with only a particular set of Kubernetes versions. For more information, refer to Astronomer's [Version Compatibility Reference](version-compatibility-reference.md).
-* We generally advise running the EKS control plane in a single security group. The worker nodes you spin up should have the same setup as the EKS control plane.
+* Astronomer recommend running the EKS control plane in a single security group. The worker nodes you spin up should have the same setup as the EKS control plane.
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/install-aws-standard.md:48]: Good catch - thanks!

File: software/install-aws-standard.md [hunk 2]
```diff
@@ -44,13 +45,13 @@ EKS is built off of Amazon's pre-existing EC2 service, so you can manage your Ku
...
 * All security and access settings needed for your worker nodes should be configured in your Cloud Formation template.
 * If you create an EKS cluster from the UI, `kubectl` access will be limited to the user who created the cluster by default.
     * To give more users `kubectl` access, you'll have to do so manually.
     * [This post](https://web.archive.org/web/20190323035848/http://marcinkaszynski.com/2018/07/12/eks-auth.html) goes through how IAM plays with EKS.
 * Expect to see each of your underlying nodes in the EC2 console.
-    * Given Astronomer's default resource request of ~11 CPUs and ~40GB of memory, we recommend using either six m5.xlarge or three m5.2xlarge [instances](https://aws.amazon.com/ec2/instance-types/) for your cluster. To modify Astronomer's default resource requests, see step 8.
+    * The default Astronomer resource request is ~11 CPUs and ~40GB of memory. Astronomer recommends using either six m5.xlarge or three m5.2xlarge [instances](https://aws.amazon.com/ec2/instance-types/) for your cluster. To modify Astronomer default resource requests, see step 8.
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/install-aws-standard.md:54]: @pgagnon correct. Updated.

File: software/install-aws-standard.md [hunk 3]
```diff
@@ -85,8 +86,9 @@ prometheus.BASEDOMAIN
 
 To obtain a TLS certificate, complete one of the following setup options:
 
-* **Option 1:** Obtain a TLS certificate from Let's Encrypt. We recommend this option for smaller organizations where your DNS administrator and Kubernetes cluster administrator are either the same person or on the same team.
-* **Option 2:** Request a TLS certificate from your organization's security team. We recommend this option for large organizations with their own protocols for generating TLS certificates.
+* **Option 1:** Obtain a TLS certificate from Let's Encrypt. Astronomer recommends this option for smaller organizations where the DNS administrator and Kubernetes cluster administrator are the same person or on the same team.
+* **Option 2:** Request a TLS certificate from your organization's security team. Astronomer recommends this option for large organizations with their own protocols for generating TLS certificates.
+* **Option 3:** Create a self-signed SSL certificate.
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/install-aws-standard.md:91]: @pgagnon I updated the title to:  _Use the AWS Certificate Manager as the certificate provider_. Let me know if this works.

File: software/install-aws-standard.md [hunk 4]
```diff
@@ -111,6 +113,26 @@ This command will generate a report. If the `X509v3 Subject Alternative Name` se
 
 Depending on your organization, you may receive either a globally trusted certificate or a certificate from a private CA. The certificate from your private CA may include a domain certificate, a root certificate, and/or intermediate certificates, all of which need to be in proper certificate order. To verify certificate order, follow the guidelines below.
 
+### Option 3: Create a self-signed SSL certificate
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/install-aws-standard.md:116]: @pgagnon see previous comment.

File: software/install-aws-standard.md [hunk 5]
```diff
@@ -111,6 +113,26 @@ This command will generate a report. If the `X509v3 Subject Alternative Name` se
...
+    ```yaml
+    nginx:
+      loadBalancerIP: ~
+      privateLoadBalancer: true  # this does affect aws-load-balancer-type: external ingressAnnotations:
+        service.beta.kubernetes.io/aws-load-balancer-type: external
+        service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: ip
+        service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "\"true\""
+        service.beta.kubernetes.io/aws-load-balancer-ssl-cert: <ACM Certificate ARN>
+        service.beta.kubernetes.io/aws-load-balancer-backend-protocol: ssl
+  ```
```
Comment (jwitz) [inline software/install-aws-standard.md:135]:
```suggestion
    ```sh
    $ openssl genrsa -out private.pem 4096  
    $ openssl rsa -in private.pem -outform PEM -pubout -out public.pem  
    ```
2.  Open the `values.yaml` file and add this entry:

    ```yaml
    nginx:
      loadBalancerIP: ~
      privateLoadBalancer: true  # this does affect aws-load-balancer-type: external ingressAnnotations:
        service.beta.kubernetes.io/aws-load-balancer-type: external
        service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: ip
        service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "\"true\""
        service.beta.kubernetes.io/aws-load-balancer-ssl-cert: <ACM Certificate ARN>
        service.beta.kubernetes.io/aws-load-balancer-backend-protocol: ssl
    ```
```

File: software/install-aws-standard.md [hunk 6]
```diff
@@ -111,6 +113,26 @@ This command will generate a report. If the `X509v3 Subject Alternative Name` se
 
 Depending on your organization, you may receive either a globally trusted certificate or a certificate from a private CA. The certificate from your private CA may include a domain certificate, a root certificate, and/or intermediate certificates, all of which need to be in proper certificate order. To verify certificate order, follow the guidelines below.
 
+### Option 3: Use the AWS Certificate Manager as the certificate provider 
+
+1. Open the `openssl` CLI and then run the following command to generate a private/public RSA key pair:
```
Comment (jwitz) [inline software/install-aws-standard.md:118]:
```suggestion
1. Run the following command to generate a private/public RSA key pair:
```
If the user has the openssl CLI installed, they shouldn't have to "open" it. We could instruct them to open a terminal window, but I think that's implied whenever we tell users to run a command.

**Review by jwitz — APPROVED** [review]
LGTM! Thanks for taking this on @kwcanuck

File: software/install-aws-standard.md (hunk 6 above)
Comment (kwcanuck) [reply software/install-aws-standard.md:118]: @jwitz the use of _open_ to orient users is accepted practice. See:

image

Typically, you use an _open_ statement to make it clear where the action occurs. If a user lands on a task, they may not always know where the action occurs, or the requirements. At Okta, we always started with _In the Admin Console_ to make sure users knew where to complete the action being described. I added this statement to make sure users understood that they were running ssl commands, but you're correct that it's probably unnecessary in this instance.

**Pull request merged**